chain/chain/src/chain.rs

```diff
@@ -4376,6 +4376,9 @@ impl<'a> ChainUpdate<'a> {
             self.chain_store_update.save_incoming_receipt(block.hash(), shard_id, receipt_proofs);
         }
 
+        self.chain_store_update.save_block_header(block.header().clone())?;
+        self.update_header_head_if_not_challenged(block.header())?;
```

Comment:
uh, this one is a little bit harder -- as here we're moving the moment when we update the 'tip', right?

Are we sure that we're not reading/depending on the tip value in the code above?

Reply:
Yes, and fortunately no.